snuba/query/validation/signature.py

```diff
@@ -104,6 +113,38 @@ def validate(self, expression: Expression, schema: ColumnSet) -> None:
             )
 
 
+class Literal(ParamType):
+    """
+    Validates that the type of a Literal expression is in a set of
+    allowed types.
+
+    If the expression provided is not a Literal, it accepts it.
+    We may consider later whether we want to enforce only literal
+    expressions can be passed as arguments in certain functions.
+    """
+
+    def __init__(self, types: Set[OptionalScalarType], allow_nullable=True) -> None:
```

Comment:
Thi should still not type check properly.
OptionalScalarType is `Union[None, bool, str, float, int, date, datetime]`.
This means that Set[Union[None, bool, str, float, int, date, datetime]] has to be a set of instances of bool, str, float, etc.

But you are calling it this way
`[("column", ColType({UInt})), ("satisfied", LiteralType({int}))],`
which means you are passing a set with the type int as element not an instance of the type int.

In order to make it type check properly your signature has to be 
`types: Set[Union[Type[int], Type[float], etc...]]`

Reply:
OK I think I figured this out now.